x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -22,9 +22,15 @@ import {
 import { i18n } from '@kbn/i18n';
 
 import { metadata } from 'ui/metadata';
+import { INDEX_PATTERN_ILLEGAL_CHARACTERS } from 'ui/index_patterns';
 
 import { CreateAnalyticsFormProps } from '../../hooks/use_create_analytics_form';
 
+// based on code used by `ui/index_patterns` internally
+// remove the space character from the list of illegal characters
+INDEX_PATTERN_ILLEGAL_CHARACTERS.pop();
+const characterList = INDEX_PATTERN_ILLEGAL_CHARACTERS.join(', ');
```

Comment:
Do you need to be checking against this list of characters, as the control for selecting the source index is a dropdown which list the Kibana index patterns, so presumably this check was done when the index pattern was created? So really the form just needs to be checking that the source index hasn't been left empty.